### PR TITLE
Add kubernetes_cluster to all kube-related events

### PR DESCRIPTION
**Pull request by awly**
Also, fix a regression in tsh, where it wouldn't update the kubeconfig
on older clusters without kubernetes_service.

Updates https://github.com/gravitational/teleport/issues/3952
Updates https://github.com/gravitational/teleport/issues/3528

### CODE REVIEW

File: lib/events/events.proto [hunk 1]
```diff
@@ -61,6 +61,11 @@ message ServerMetadata {
     // session occurred on.
     map<string, string> ServerLabels = 5
         [ (gogoproto.nullable) = false, (gogoproto.jsontag) = "server_labels,omitempty" ];
+
+    // Kubernetes has information about a kubernetes cluster, if this was a
+    // kubernetes server request.
+    KubernetesClusterMetadata Kubernetes = 6
```
Comment (r0mant) [inline lib/events/events.proto:67]: I'm not sure TBH that "server metadata" is the best place to add Kubernetes cluster name... As I understand it (and I may be mistaken), it is supposed to contain information about particular cluster node you're teleporting onto, while K8s cluster name is something "global" to a cluster, outside of a single server scope.

Have you considered adding `KubernetesMetadata` as a separate field to those session events? That's what AAP did I think (and I'm doing the same for db).

Comment (awly) [reply lib/events/events.proto]: Makes sense, moved to a separate field.

**Review by r0mant — APPROVED** [review]

File: lib/events/events.proto [hunk 2]
```diff
@@ -99,6 +106,10 @@ message SessionStart {
 
     // TerminalSize is expressed as 'W:H'
     string TerminalSize = 6 [ (gogoproto.jsontag) = "size,omitempty" ];
+
+    // Kubernetes has information about a kubernetes cluster, if applicable.
```
Comment (a-palchikov) [inline lib/events/events.proto:110]: Kubernetes -> KubernetesClusterMetadata

File: lib/events/events.proto [hunk 3]
```diff
@@ -122,6 +133,10 @@ message SessionJoin {
     // ConnectionMetadata holds information about the connection
     ConnectionMetadata Connection = 5
         [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // Kubernetes has information about a kubernetes cluster, if applicable.
```
Comment (a-palchikov) [inline lib/events/events.proto:137]: Kubernetes -> KubernetesClusterMetadata
and below

File: lib/kube/kubeconfig/kubeconfig.go [hunk 4]
```diff
@@ -108,6 +107,13 @@ func UpdateWithClient(ctx context.Context, path string, tc *client.TeleportClien
 		if err != nil && !trace.IsNotFound(err) {
 			return trace.Wrap(err)
 		}
+
+		// If there are no registered k8s clusters, we may have an older
+		// teleport cluster. Fall back to the old kubeconfig, with static
+		// credentials from v.Credentials.
+		if len(v.Exec.KubeClusters) == 0 {
+			v.Exec = nil
```
Comment (a-palchikov) [inline lib/kube/kubeconfig/kubeconfig.go:115]: Leave a debug log about it?

**Review by a-palchikov — APPROVED** [review]

File: lib/events/events.proto (hunk 2 above)
Comment (awly) [reply lib/events/events.proto:110]: `Kubernetes` is the field name, i'll leave as is

File: lib/kube/kubeconfig/kubeconfig.go (hunk 4 above)
Comment (awly) [reply lib/kube/kubeconfig/kubeconfig.go:116]: added

**Pull request merged**